src/etos_lib/kubernetes/etos.py

```diff
@@ -84,11 +83,27 @@ class Kubernetes:
     __namespace = None
     logger = logging.getLogger(__name__)
 
-    def __init__(self, version="v1alpha1"):
+    def __init__(self, version="v1alpha1", config_path: Union[None, str, Path] = None):
         """Initialize a dynamic client with version."""
+        self.load_kubernetes_config(config_path)
         self.version = f"etos.eiffel-community.github.io/{version}"
         self.__client = DynamicClient(api_client.ApiClient())
 
+    def load_kubernetes_config(self, path: Union[None, str, Path]):
+        """Load a Kubernetes config if possible, will log an error if not possible.
+
+        If the config is not loaded properly then the Kubernetes client will fail in errors
+        later. We catch the exception instead of raising it up the stack so that tests
+        are easier to write without having to catch Exceptions in them.
+        """
+        try:
+            if path is not None:
+                load_config(kube_config_path=str(path))
+            else:
+                load_config()
+        except ConfigException:
+            self.logger.exception("Failed to load Kubernetes config. The client won't work.")
```

Comment:
You could add OTEL instrumentation (span and/or event) here to make it more visable when things go south.

Reply:
Adding the event here would that much since we typically have not started a span when this class is initialized. When the service later uses the client it would raise an exception that the config has not been loaded and this should happen within a span context manager and an automatic event would be sent with proper links to a trace.